docs/content/en/docs/concepts/architecture/components/scheduler/_index.md

```diff
@@ -14,3 +14,152 @@ The Lifecycle Scheduler can trigger events such as
 the Pre-Deployment checks are completed.
+
+## Architecture of Keptn Scheduler
+
+To integrate the `Keptn Lifecycle Scheduler` into a Kubernetes cluster,
+a Kubernetes manifest is applied with specific **Keptn annotations**.
+These annotations identify the workload and specify the
+pre-deployment and post-deployment tasks and evaluations
+associated with it.
+
+## How it works
+
+The `Keptn Scheduler`, integrated into the `Keptn Lifecycle Operator`,
+which is equipped with multiple controllers for managing
+**Keptn Custom Resource Definitions (CRDs)** and a powerful
+**Mutating Webhook**.
+Its core responsibility is to orchestrate the scheduling of Pods, taking
+into account pre-deployment checks to ensure optimal deployment success.
+
+Let's dive into its architectural details:
+
+- **Scheduler Flow**:
+After applying a Kubernetes Manifest annotated with Keptn-specific
+annotations to the cluster, the Mutating Webhook injects the Keptn Scheduler.
+The Scheduler then receives Kubernetes Events for
+Pre-Deployment, which are sent to the event stream.
+The Event Controller watches for these events and triggers a
+Kubernetes Job to fulfill the Pre-Deployment tasks.
+Once the Pre-Deployment phase is complete,
+the Keptn Scheduler schedules the Pod for deployment.
+
+- **Controllers**:
+The Keptn Lifecycle Operator contains several controllers
+for `Keptn Custom Resource Definitions (CRDs)` and a `Mutating Webhook`.
+These controllers, including **KeptnApp** and **KeptnWorkload**,
+monitor and manage workload resources and checks associated with an application.
+
+    ```yaml
+    apiVersion: v1
+    kind: Namespace
+    metadata:
+      name: podtato-kubectl
+      annotations:
+        keptn.sh/lifecycle-toolkit: "enabled"  # this lines tells the webhook to handle the    namespace
+        keptn.sh/workload: "some-workload-name"
+        keptn.sh/app: "your-app-name"
+    ```
+
+- **KeptnApp**:
+The [KeptnApp](https://lifecycle.keptn.sh/docs/yaml-crd-ref/app/) represents information about all workloads
+and checks associated with an application.
+It specifies the pre/post deployment and evaluation checks
+to be executed at the app level.
+Changing the app version triggers new executions of
+app-level checks, while changes in the workload version
+impact only workload checks.
+
+    ```yaml
+    apiVersion: lifecycle.keptn.sh/v1alpha3
+    kind: KeptnApp
+    metadata:
+      name: podtato-head
+      namespace: podtato-kubectl
+    spec:
+      version: "1.3"
+      revision: 1
+      workloads:
+        - name: podtato-head-left-arm
+          version: 0.1.0
+        - name: podtato-head-left-leg
+          version: 1.2.3
+      preDeploymentTasks:
+        - pre-deployment-hello
+      postDeploymentTasks:
+        - post-deployment-hello
+      preDeploymentEvaluations:
+        - my-prometheus-definition
+      postDeploymentEvaluations:
+        - my-prometheus-definition
+    ```
+
+- **KeptnWorkload**:
```

Comment:
For this section, each resource should be a link to the reference page in the https://main.lifecycle.keptn.sh/docs/yaml-crd-ref/ section or, if it is not there, in the https://main.lifecycle.keptn.sh/docs/crd-ref/ section.